cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/AppDeploymentInfo.java

```diff
@@ -70,6 +73,8 @@ public AppDeploymentInfo(ArtifactId artifactId, Location artifactLocation, Names
     this.configString = configString;
     this.ownerPrincipal = ownerPrincipal;
     this.updateSchedules = updateSchedules;
+    this.isRuntime = isRuntime;
+    this.userProps = userProps;
```

Comment:
nit: Assert that the `userProps` is not null. Otherwise, assume that `getUserProps` can be nullable.

Reply:
`userProps` won't be null, it will be empty if nothing is in user properties.